File: python/1975-Maximum Matrix Sum.py
import numpy as np

class Solution(object):
  def maxMatrixSum(self, matrix):
    """
    :type matrix: List[List[int]]
    :rtype: int
    """
    numNegativeValues = 0
    minVal = abs(matrix[0][0])
    absSumMatrix = 0
    for i in range(len(matrix)):
      for j in range(len(matrix[i])):
        if matrix[i][j] < 0:
          numNegativeValues += 1

        thisAbs = abs(matrix[i][j])
        absSumMatrix += abs(matrix[i][j])
        #print(thisAbs, minVal)
        if thisAbs < minVal:
          minVal = thisAbs
    
    #print(absSumMatrix)    
    if 1 == numNegativeValues % 2:
      absSumMatrix -= 2 * minVal
    #print(numNegativeValues, absSumMatrix, minVal)
    
    return absSumMatrix

File: cpp/1000 to 1999/1975-Maximum Matrix Sum.cpp
#include <vector>
#include <iostream>

class Solution
{
public:
  long long maxMatrixSum(vector<vector<int>>& matrix)
  {
    int numNegativeValues = 0;
    int absMinVal = std::abs(matrix[0][0]);
    long long absSumMatrix = 0;
    for(size_t i = 0; i < matrix.size(); i++)
    {
      for(size_t j = 0; j < matrix[i].size(); j++)
      {
        if(matrix[i][j] < 0)
        {
          numNegativeValues++;
        }
        int thisAbsVal = std::abs(matrix[i][j]);
        absSumMatrix += thisAbsVal;
        absMinVal = std::min(absMinVal, thisAbsVal);
      }
    }

    if(1 == numNegativeValues % 2)
    {
      absSumMatrix -= 2 * absMinVal;
    }

    return absSumMatrix;
  }
};